src/Intersection.hs

```diff
@@ -0,0 +1,18 @@
+module Intersection where
+
+import Data.Ord
+
+import DataTypes
+
+data Intersection = Intersection {
+  intersectedObject :: Object,
+  distance :: Double
+} deriving Eq
+
+instance Ord Intersection where
+  compare x y
+    | distance x < distance y = LT
+    | distance x > distance y = GT
+    | otherwise = EQ
+  x <= y = distance x <= distance y
```

Comment:
Cool! Interesting that you define less than and equal to though, i suppose it's the only thing that's not covered by those cases, and greater than equal is just the opposite case.

not bad haskell!